File: app/src/main/java/com/example/cykashoppinglist/util/DateParser.java
package com.example.cykashoppinglist.util;

import android.annotation.SuppressLint;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class DateParser {
	@SuppressLint("SimpleDateFormat")
	public static Date parse(String input) {
		input = input.replace("T", " ");
		DateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
		try {
			return format.parse(input);
		} catch (ParseException e) {
			e.printStackTrace();
			return new Date();
		}
	}

	public static String parse(Date input) {
		DateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
		String output = format.format(input);
		output = output.replace(" ", "T");
		return output;
	}
}

File: app/src/main/java/com/example/cykashoppinglist/util/DateParser.kt
package com.example.cykashoppinglist.util

import android.annotation.SuppressLint
import java.text.DateFormat
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

object DateParser {
	@SuppressLint("SimpleDateFormat")
	fun parse(input: String): Date {
		var input = input
		input = input.replace("T", " ")
		val format: DateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
		return try {
			format.parse(input)
		} catch (e: ParseException) {
			e.printStackTrace()
			Date()
		}
	}

	fun parse(input: Date?): String {
		val format: DateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
		var output = format.format(input)
		output = output.replace(" ", "T")
		return output
	}
}